File: frontend/src/types/playlist.ts
import type { Track } from './songs';

export interface Playlist {
    id: number;
    name: string;
    description: string;
    tracks: Track[];
}

export interface NewPlaylist {
    name: string;
    description: string;
}

export interface PlaylistFilters {
    name?: string;
    nameOrder?: boolean | null;
    description?: string;
    descriptionOrder?: boolean | null;
}

File: backend/Models/Playlist.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models;

public class PlaylistCreate
{
    public string Name { get; set; }
    public string Description { get; set; }
    public bool SoftDelete { get; set; }
}

public class PlaylistRead : PlaylistCreate
{
    [Key]
    public int Id { get; set; }
}
public class Playlist : PlaylistRead
{
    public User User { get; set; }
    [ForeignKey("User")]
    public int UserId { get; set; }

    public List<Track> Tracks { get; set; }
}